### PR TITLE
Factors: Add support for negative numbers. Add negative cases and tidy (#4245)

### LINKED ISSUE #4245: Factors: Negatives will not appear
If I search `factors of 7`, factors will appear. However, no Instant Answers will appear when I search `factors of -7`.

------
IA Page:  http://duck.co/ia/view/factors 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @austinheimark

### PR DESCRIPTION
**Pull request by g33kgrrl**
## Description of new Instant Answer, or changes

Factors: Add support for factoring negative numbers. Also add negative cases to test, and tidy code.

Fixes #4245. For thoroughness' sake, first confirmed that the mathematically correct process for finding factors for a negative number involves the following steps:
* Find positive factors
* Duplicate positive factors and change sign
* Sort in ascending order
* Concatenate with sorted positive factors

Implemented function as described above. Also, there were no negative cases in the test, so added two negative tests according to testing standards. Tidied code with perltidy.

## People to notify

@pjhampton @austinheimark @keeganjk

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/factors

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by pjhampton** [comment]
Hey @g33kgrrl 

Welcome! Thanks for your contribution. Can you please update the bottom of the issues template with the IA id?

**Comment by g33kgrrl** [comment]
Hi @pjhampton,

Ah yes, I see I missed one at the bottom.  Corrected, thanks!

File: lib/DDG/Goodie/Factors.pm [hunk 1]
```diff
@@ -13,18 +14,27 @@ triggers startend => 'factors', 'factors of';
 
 handle remainder => sub {
     my $query = $_;
-    return unless $query =~ /^\d+$/;
 
-    my $factors = join ', ', divisors($query);
+    return unless $query =~ /^-?\d+$/;
```
Comment (mintsoft) [inline lib/DDG/Goodie/Factors.pm]: You `could` do this in one step if you wanted with some regex captures:
```
return unless $query =~ /^(-)?(\d+)$/;
```
then you'd have `$1` and `$2` which are `-`  and the number; or you could use named captures if you wanted it to be more explicit like:

```
return unless $query =~ /^(?<neg>-)?(?<num>[0-9]+)$/;
# here you have $+{'neg'} and $+{'num'}
```

Comment (g33kgrrl) [inline lib/DDG/Goodie/Factors.pm]: @mintsoft True, thanks for the suggestion! I think the latter option might be a little too cryptic, but I like the efficiency of the former approach. I'll get that going and post an update.

**Comment by g33kgrrl** [comment]
Done!  Please let me know if you spot any other issues.

**Comment by bsstoner** [comment]
This was released to production 🚀
